src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1346,10 +1348,21 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                 BoundStatement bodyWithoutAsync = AsyncRewriter.Rewrite(bodyWithoutIterators, method, methodOrdinal, lazyVariableSlotAllocator, compilationState, diagnostics,
                     out AsyncStateMachine asyncStateMachine);
 
+                if (bodyWithoutAsync.HasErrors)
+                {
+                    return bodyWithoutAsync;
+                }
+
+                BoundStatement bodyWithNullChecks = bodyWithoutAsync;
+                if (sawNullChecked)
+                {
+                    bodyWithNullChecks = (BoundStatement)NullCheckRewriter.Rewrite(bodyWithoutAsync, method, methodOrdinal, lazyVariableSlotAllocator, compilationState, diagnostics);
```

Comment:
>Rewrite [](start = 75, length = 7)

Consider changing `Rewrite()` to return a `BoundStatement` so the caller does not need to assume that.